plugins/org.locationtech.udig.image.georeferencing/src/org/locationtech/udig/image/georeferencing/internal/ui/coordinatepanel/CoordinateTableComposite.java

```diff
@@ -79,1132 +69,1171 @@
-	 * @param style
-	 *            The style.
-	 */
-	public CoordinateTableComposite(GeoReferencingCommand cmd, Composite parent, int style) {
+    private ImageRegistry registry = null;
 
-		super(parent, style);
+    private List<MouseSelectionListener> listeners = new LinkedList<>();
 
-		assert cmd != null;
-		this.cmd = cmd;
-		createContent();
-	}
+    private ToolItem itemDeleteAll;
 
-	private void createContent() {
+    /**
+     * Constructor method.
+     *
+     * @param cmd The GeoReferencingCommand.
+     * @param parent The parent composite.
+     * @param style The style.
+     */
+    public CoordinateTableComposite(GeoReferencingCommand cmd, Composite parent, int style) {
 
-		this.registry = createImageRegistry();
-		this.uiThread = Thread.currentThread();
+        super(parent, style);
 
-		createListeners();
-		setCoordinateTools();
+        assert cmd != null;
+        this.cmd = cmd;
+        createContent();
+    }
 
-		GridLayout gridLayout2 = new GridLayout();
-		gridLayout2.numColumns = 2;
-		gridLayout2.makeColumnsEqualWidth = true;
+    private void createContent() {
 
-		compositeGridCoord = new Composite(this, SWT.BORDER);
-		compositeGridCoord.setLayout(gridLayout2);
+        this.registry = createImageRegistry();
+        this.uiThread = Thread.currentThread();
 
-		createCRSselector(compositeGridCoord);
-		createToolbar(compositeGridCoord);
-		createGrid(compositeGridCoord);
-	}
+        createListeners();
+        setCoordinateTools();
 
-	/**
-	 * Creates the required listeners, the majority of them are tool listeners.
-	 */
-	private void createListeners() {
+        GridLayout gridLayout2 = new GridLayout();
+        gridLayout2.numColumns = 2;
+        gridLayout2.makeColumnsEqualWidth = true;
 
-		this.capturedListener = new CapturedCoordinateListener() {
+        Composite compositeGridCoord = new Composite(this, SWT.BORDER);
+        compositeGridCoord.setLayout(gridLayout2);
 
-			public void capturedCoordinate(Coordinate newCoord) {
+        createCRSselector(compositeGridCoord);
+        createToolbar(compositeGridCoord);
+        createGrid(compositeGridCoord);
+    }
 
-				addCoordinateFromMapToTable(newCoord);
+    /**
+     * Creates the required listeners, the majority of them are tool listeners.
+     */
+    private void createListeners() {
 
-				assert getMainComposite().getMapMarkGraphic() != null;
-				getMainComposite().refreshMapGraphicLayer();
-			}
+        this.capturedListener = new CapturedCoordinateListener() {
 
-			public void activated(boolean active) {
-			}
-		};
+            @Override
+            public void capturedCoordinate(Coordinate newCoord) {
 
-		this.deletedListener = new DeletedCoordinateListener() {
+                addCoordinateFromMapToTable(newCoord);
 
-			public void deletedCoordinate(java.awt.Point point, InputEvent event) {
+                MainComposite mainComposite = getMainComposite();
+                assert mainComposite != null;
```

Comment:
I guess this is the fix for potential NPE, quite hard to find due to re-format of code

Reply:
There are many many rows like this which are changed. Hard to see while formatting, indeed.